### PR TITLE
4895

### PR DESCRIPTION
**Pull request by m-nash**

### CODE REVIEW

File: azure-proto-authorization/RoleAssignmentOperations.cs [hunk 1]
```diff
@@ -48,6 +48,8 @@ internal RoleAssignmentOperations(AzureResourceManagerClientOptions options, Res
 
         private RoleAssignmentsOperations Operations => GetClient<AuthorizationManagementClient>((baseUri, creds) => new AuthorizationManagementClient(Id.Subscription, baseUri, creds)).RoleAssignments;
 
+        protected override ResourceType ValidResourceType => ResourceType;
```
Comment (allenjzhang) [inline azure-proto-authorization/RoleAssignmentOperations.cs:51]: nit, I think the protected needs to be above private.

**Comment by allenjzhang** [comment]
>         internal AvailabilitySetsOperations Operations => GetClient<ComputeManagementClient>((uri, cred) =>

private?  Few other places are private. I think I have updated some in the core to private as well.

---
Refers to: azure-proto-compute/AvailabilitySetOperations.cs:108 in bbf18c1. [](commit_id = bbf18c1d5e163d593ff6609104f0f94d73152dca, deletion_comment = False)

File: azure-proto-compute/VirtualMachineOperations.cs [hunk 2]
```diff
@@ -141,7 +246,14 @@ public async Task<ArmOperation<VirtualMachine>> UpdateAsync(VirtualMachineUpdate
...
         }
 
-        public ArmOperation<VirtualMachine> AddTag(string key, string value)
+        /// <summary>
+        /// Add a tag to a virtual machine.
+        /// If the tag already exists it will be modified.
+        /// </summary>
+        /// <param name="key"> The key for the tag. </param>
+        /// <param name="value"> The value for the tag. </param>
+        /// <returns> An <see cref="ArmOperation{VirtualMachine}"/> that allows polling for completion of the operation. </returns>
```
Comment (allenjzhang) [inline azure-proto-compute/VirtualMachineOperations.cs:255]: <remarks> on LRO?

File: azure-proto-compute/Extensions/AzureSubscriptionExtension.cs [hunk 3]
```diff
@@ -45,14 +45,14 @@ public static AsyncPageable<VirtualMachine> ListVirtualMachinesAsync(this Subscr
 
         public static Pageable<ArmResource> ListVirtualMachinesByName(this SubscriptionOperations subscription, ArmSubstringFilter filter, int? top = null, CancellationToken cancellationToken = default)
         {
-            ArmFilterCollection filters = new ArmFilterCollection(VirtualMachineData.ResourceType);
+            ArmFilterCollection filters = new ArmFilterCollection(VirtualMachineOperations.ResourceType);
```
Comment (allenjzhang) [inline azure-proto-compute/Extensions/AzureSubscriptionExtension.cs:48]: What do you think of the idea of placing ResourceType on VirtualMachine?  Seems more natural.

File: azure-proto-compute/Placeholder/VirtualMachineData.cs [hunk 4]
```diff
@@ -5,10 +5,15 @@
...
 {
+    /// <summary>
+    /// A class representing the VirtualMachine data model.
+    /// </summary>
     public class VirtualMachineData : TrackedResource<Azure.ResourceManager.Compute.Models.VirtualMachine>
     {
-        public static ResourceType ResourceType => "Microsoft.Compute/virtualMachines";
-
+        /// <summary>
+        /// Initializes a new instance of the <see cref="______"/> class.
```
Comment (allenjzhang) [inline azure-proto-compute/Placeholder/VirtualMachineData.cs:14]: nit, ___ -> VirtualMachineData

File: Azure.ResourceManager.Core/SubscriptionOperations.cs [hunk 5]
```diff
@@ -48,15 +48,13 @@ internal SubscriptionOperations(AzureResourceManagerClientOptions options, Resou
...
         /// <summary>
         /// Gets the subscription client.
         /// </summary>
-        internal SubscriptionsOperations SubscriptionsClient => GetClient((uri, cred) =>
-            new ResourcesManagementClient(
-                uri,
-                Guid.NewGuid().ToString(),
-                cred,
-                ClientOptions.Convert<ResourcesManagementClientOptions>())).Subscriptions;
+        internal SubscriptionsOperations SubscriptionsClient => GetClient<ResourcesManagementClient>((uri, cred) =>
```
Comment (allenjzhang) [inline Azure.ResourceManager.Core/SubscriptionOperations.cs:56]: Resharper actually shows the GetClient<Type> is redundant. Thus I removed it. Also multiline is earlier to read.

**Review by allenjzhang — APPROVED** [review]
:shipit:

File: azure-proto-compute/Extensions/AzureSubscriptionExtension.cs (hunk 3 above)
Comment (m-nash) [reply azure-proto-compute/Extensions/AzureSubscriptionExtension.cs:48]: I have that as a backlog item to consider

File: azure-proto-authorization/RoleAssignmentOperations.cs (hunk 1 above)
Comment (m-nash) [reply azure-proto-authorization/RoleAssignmentOperations.cs:51]: other way around

**Pull request merged**